tests/integration/issues/github_2992/test_flow.py

```diff
@@ -0,0 +1,18 @@
+from jina import Document, Flow
+from jina.types.arrays.memmap import DocumentArrayMemmap
+
+
+def test_dam_flow(tmpdir):
+    def _assert_called_once(response):
+        assert len(response.docs) == 1
+        assert not _assert_called_once.called
+        _assert_called_once.called = True
+
+    _assert_called_once.called = False
+
+    f = Flow().add()
+    dam = DocumentArrayMemmap(tmpdir)
+    dam.append(Document())
+    with f:
+        response = f.post('/', dam, on_always=_assert_called_once, return_results=True)
```

Comment:
remove the `on_always` and the callback it is not needed anymore

Reply:
but the issue is that in a failing case, the interpreter keeps processing f.post forever. This means the response will never be used. I think on_always is the best way to detect the bug efficiently because if on_always is called more than once it means there's a bug.

Reply:
if u check that only one response is processed then it is enough. Having callbacks as tests is hard since very little guarantee that the assert is called. We have seen this problem before

Reply:
Honestly I don't know a way to catch the bug (infinitely posting documents, without ever existing the f.post line) without using on_always. Can you help with that ? Maybe there's a timeout parameter that I can use ?

Reply:
responses are not returned until all the requests are handled, so the bug is already caught like this

Reply:
So passing this tedt without the callback already guarantees the fixing of the bug. U can try without the fix and u will see